### ISSUE TITLE
File browser: UTF-8 strings not respected in Patient's Name

### ISSUE THREAD

**Issue by mahanako** [innolitics/dicom-standard#84]
**Bug description**
UTF-8 characters are not respected in the Patient's Name

**Steps for reproduction**
Upload a DICOM file that contains a Japanese (eg. 鈴木), Chinese (eg. 王) or Korean (eg. 김) Name or any Latin name containing diacritics like ÄÖÜßÑ.. 
=> Name characters are garbled

**Expected behavior**
The Patient's Name displayed correctly

**Comment by mahanako** [comment]
Even with Specific Character Set (0008,0005) being set to ISO_IR 192, which would state that UTF-8 is being used.

**Comment by MichaelLeeHobbs** [comment]
# Response to #84 — UTF-8 strings not respected in Patient's Name

Confirmed by the follow-up: the file declares Specific Character Set (0008,0005) = `ISO_IR 192`, which is UTF-8 (PS3.3 Table C.12-2). A reader garbling those names is ignoring (0008,0005).

Never a one-line fix, for the record: (0008,0005) is multi-valued for the ISO 2022 escape-sequence sets, and Person Name has three `=`-delimited component groups that may each use a different set. Handling `ISO_IR 192` alone would still mishandle #77. PS3.5 §6.1.2–6.2.

**Valid, but recommend closing with the File Editor beta — see #96.**

*(Drive-by triage, not a maintainer.)*

**Comment by mahanako** [comment]
Thanks for the reply and confirming the bug's validity. https://github.com/innolitics/dicom-standard/issues/96 somehow only states to discontinue a file editor beta. The problem I've described has nothing to do with editing or do you plan to discontinue the functionality to verify standard conformity of files altogether?

**Comment by MichaelLeeHobbs** [comment]
@mahanako I am not the maintainer. I saw that the maintainer was overloaded with issues. I use my deep experience with DICOM to review and check all of the open issues. Yes, your bug is valid, but I do not recommend fixing it, as no one should upload non-anonymized DICOM to any public website, regardless of what the site says about not using the data. Would you upload your government ID to a website, even if they say it's only in your browser? There are plenty of offline tools for this sort of stuff. That is why I recommend discontinuing the functionality of uploading for reviewing/editting.

**Comment by mahanako** [comment]
You're right, I would not upload actual data, let alone non-anonymized. And I never did upload actual data. I do upload fantasy data, for which I am grateful that the website tells me whether the data I've generated is formally correct. Unfortunately my fantasy names got garbled, hence the bug - no real data in play at any time.

Will the service of checking files against the DICOM standard cease to exist?

**Comment by MichaelLeeHobbs** [comment]
@mahanako To be clear, I'm not the maintainer, so this isn't my decision to make. My recommendation is to discontinue the feature: nearly half of the open bugs relate to it, and it encourages people to upload real DICOM data.

**Comment by mahanako** [comment]
Don't worry, I know that. Thanks for replying. Means a lot :)

**Comment by johndgiese** [comment]
Hey guys, this is the maintainer. I'd note the DICOM analysis function is done fully in the browser, so the idea is it would be fine for people to upload real data.

**Comment by mahanako** [comment]
Hi @johndgiese, 

Thanks for your reply. I've just tried uploading my file again and the UTF-8 encoded Patient Name is still garbled.
<img width="1410" height="306" alt="Image" src="https://github.com/user-attachments/assets/e048c894-cae2-4d07-9b11-2ca7bf695b87" />

**Comment by mahanako** [comment]
@MichaelLeeHobbs 
> There are plenty of offline tools for this sort of stuff.

With your deep understanding of DICOM, could you name a few of those tools? I myself use Offis's DCMTK library to write out data, using this website to validate the write-out's conformity.

**Comment by MichaelLeeHobbs** [comment]
@mahanako 

Check my repo: https://github.com/MichaelLeeHobbs/awesome-hl7-dicom

Here are a few quick ones that all do similar things, but slightly differently.
https://weasis.org/en/tutorials/dicomizer/
https://www.microdicom.com/
https://www.dvtk.org/dicom/editor/
https://dicom.offis.de/en/dcmtk/dcmtk-tools/

**Comment by mahanako** [comment]
@MichaelLeeHobbs 
Thank you! I'll also have a look at your repository. I do use Weasis's viewer - that one interpreted UTF-8 strings correctly.
